api/v1/db/json/notes-dao.js

```diff
@@ -191,4 +197,22 @@ const patchNoteByID = (noteID, body) => new Promise((resolve, reject) => {
   }
 });
 
-module.exports = { getNotes, postNotes, patchNoteByID };
+const deleteNoteByID = noteID => new Promise((resolve, reject) => {
+  try {
+    const rawNote = fetchNote(noteID);
+    if (!rawNote) {
+      resolve(null);
+    }
+
+    const studentID = noteID.split('-')[0];
+    const noteFilePath = `${dbDirectoryPath}/${studentID}/${noteID}.json`;
+    fs.unlinkSync(noteFilePath);
```

Comment:
What's the behavior if the `noteFilePath` doesn't exist? Will `fs` handle it properly? Like we discussed in person, it will be more organized if we put all `fs` related method and their handlers into a class.

Reply:
Previously, this was handled by first trying to fetch the note and checking if the result was null. Now, this is handled by the method in `fs-operations.js`.